InMobi/src/main/java/com/mopub/mobileads/InMobiRewarded.kt

```diff
@@ -0,0 +1,198 @@
+package com.mopub.mobileads
+
+import android.app.Activity
+import android.content.Context
+import com.inmobi.ads.AdMetaInfo
+import com.inmobi.ads.InMobiAdRequestStatus
+import com.inmobi.ads.InMobiInterstitial
+import com.inmobi.ads.exceptions.SdkNotInitializedException
+import com.inmobi.ads.listeners.InterstitialAdEventListener
+import com.inmobi.sdk.InMobiSdk
+import com.mopub.common.DataKeys
+import com.mopub.common.LifecycleListener
+import com.mopub.common.MoPubReward
+import com.mopub.common.logging.MoPubLog
+import com.mopub.common.logging.MoPubLog.AdapterLogEvent
+import com.mopub.mobileads.InMobiAdapterConfiguration.Companion.onInMobiAdFailWithError
+import com.mopub.mobileads.InMobiAdapterConfiguration.Companion.onInMobiAdFailWithEvent
+import java.lang.Error
+
+open class InMobiRewarded : BaseAd() {
```

Comment:
`open` has to be removed. And I would still suggest to either extend `InMobiInterstitial` or create some common file and keep everything common in that file for both Rewarded and Interstitial.

This is something I would not encourage even if it takes time in PR review or merge. If there are any time constraints then create JIRA and comment this on the PR that you're going to open of Mopub to also keep them in sync otherwise they'll ask why it was not done in the first place.

Reply:
Done https://jira.corp.inmobi.com:8443/browse/DI-14521